src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3479,24 +3478,31 @@ public Object stringToF(DynamicObject string, boolean strict,
     @ImportStatic(StringGuards.class)
     public static abstract class StringIndexPrimitiveNode extends CoreMethodArrayArgumentsNode {
 
-        @Child private StringByteCharacterIndexNode byteIndexToCharIndexNode = StringNodesFactory.StringByteCharacterIndexNodeFactory.create(null);
-        @Child private NormalizeIndexNode normalizeIndexNode = StringNodesFactory.NormalizeIndexNodeGen.create(null, null);
+        @Child private EncodingNodes.CheckEncodingNode checkEncodingNode;
+
+        @Specialization(guards = "start < 0")
+        public Object stringIndexBadStart(DynamicObject string, DynamicObject pattern, int start) {
+            throw new RaiseException(getContext().getCoreExceptions().argumentError("negative start given", this));
+        }
+
+        @Specialization(guards = { "start >= 0", "isEmpty(pattern)" })
+        public Object stringIndexEmptyPattern(DynamicObject string, DynamicObject pattern, int start) {
+            return start;
+        }
```

Comment:
Does this happen in practice? Or can we just assert against it?

Reply:
The string can be empty in practice.